xarray/conventions.py

```diff
@@ -369,10 +375,17 @@ def __init__(self, array, units, calendar=None):
         self.array = array
         self.units = units
         self.calendar = calendar
-        # Verify at least one date can be decoded successfully.
-        # Otherwise, tracebacks end up swallowed by Dataset.__repr__ when users
-        # try to view their lazily decoded array.
-        example_value = first_n_items(array, 1) or 0
+
+        # Verify that at least the first and last date can be decoded 
+        # successfully. Otherwise, tracebacks end up swallowed by 
+        # Dataset.__repr__ when users try to view their lazily decoded array.
+        example_value = first_n_items(array, 1) or [0]
+
+        if array.size > 1:
+            # fixes (part of) https://github.com/pydata/xarray/issues/975
+            example_value_end = last_item(array)
+            example_value = np.concatenate((example_value, example_value_end))
```

Comment:
Rather than going through all these conditions, I would just construct this once, e.g.,
```
example_values = np.concatenate([first_n_items(array, 1), last_item(array), [0]])
```

It's not very expensive to try decoding even three values.